### PR TITLE
fix(anthos-attached-clusters): add providers and lint

### PR DESCRIPTION
**Pull request by apeabody**
#### Description
- Add any missing providers and versions constraints
- Additional lint fixes

### CODE REVIEW

File: anthos-attached-clusters/aks/provider.tf [hunk 1]
```diff
@@ -25,7 +25,8 @@ terraform {
       version = ">=5.0.0"
     }
     helm = {
-      source = "hashicorp/helm"
+      source  = "hashicorp/helm"
+      version = "~> 2.0"
```
Comment (NimJay) [inline anthos-attached-clusters/aks/provider.tf:29]: **Praise:** Good idea adding version constraints.

File: anthos-attached-clusters/aks/main.tf [hunk 2]
```diff
@@ -20,9 +20,6 @@ locals {
   }
 }
 
-data "azurerm_subscription" "current" {
-}
-
```
Comment (NimJay) [inline anthos-attached-clusters/aks/main.tf:25]: **Thought:** Judging from [this search result](https://github.com/search?q=repo%3AGoogleCloudPlatform%2Fanthos-samples%20azurerm_subscription&type=code), there seems to be another unused instance of `data "azurerm_subscription" "current"` at: https://github.com/GoogleCloudPlatform/anthos-samples/blob/ef3baae55c6ec0e98cc062bfcfbda26345e0fe15/anthos-multi-cloud/Azure/modules/cluster-rg/main.tf#L24

Remove that is out-of-scope for this PR (but feel free to include it).

**Review by NimJay — APPROVED** [review]
Thanks again, @apeabody! Approved. Feel free to merge whenever you're comfortable!

File: anthos-attached-clusters/aks/main.tf (hunk 2 above)
Comment (apeabody) [reply anthos-attached-clusters/aks/main.tf:25]: Agreed - To keep the PR/review scope reasonable, I plan to do a similar PR for `anthos-multi-cloud`.

**Pull request merged**